### PR TITLE
Feat: Admin managing posts feature added successfully issue 339

### LINKED ISSUE #339: [Feature Request]: Creating manging post by admin created by users
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

This give admin access to delete and edit a users post from the webapp.
In case of spam/malicious it can be prevent so that furthur damage can be minimized easily.

### Use Case

Webapp content management

### Benefits

Spam management

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by MeetDOD**
# Pull Request Resolves [#339 ]

### Description

1. In Backend, created 3 controllers for editing, delete post and getpostbyid and their controllers respectively.
2. In Frontend, creating a new route for update the post, deleting the post by delete api and updating first by fetching the data from id getpostbyid and updating it and navigate back to posts page.
3. Used appropriate toast notification.
4. The design is responsive for all the devices.

### Related Issues
Fixes #339 

### Changes Made
New feature: Admin managing posts feature added.

### Screenshots

image

image

image

### Video

https://github.com/VaibhavArora314/StyleShare/assets/120380073/5c628c94-eb1c-45fb-89d2-a828755473fd

I certify that I have carried out the relevant code of conduct and provided the requisite screenshot for validation by submitting this pull request.

Thank You for this contribution.

### CODE REVIEW

File: backend/src/routes/admin/controller.ts [hunk 1]
```diff
@@ -232,3 +232,148 @@ export const getAdminStatsController = async (req: Request, res: Response) => {
...
+        isAdmin: true,
+      },
+    });
+
+    if (!user?.isAdmin) {
+      return res.status(403).json({
+        error: "User is not admin!",
+      });
+    }
+
```
Comment (VaibhavArora314) [inline backend/src/routes/admin/controller.ts:295]: This user admin check is not needed since it is already done in isAdmin middleware

**Review by VaibhavArora314 — CHANGES_REQUESTED** [review]

File: backend/src/routes/admin/controller.ts (hunk 1 above)
Comment (MeetDOD) [reply backend/src/routes/admin/controller.ts]: @VaibhavArora314 I was also not doing this but I taught lets do like the user post contoller approach, My bad but i removed that check now 👍

**Review by VaibhavArora314 — APPROVED** [review]

**Review by Ultimateutkarsh11 — APPROVED** [review]

**Comment by MeetDOD** [comment]
@VaibhavArora314 can you add the label and level please !

**Review by akbatra567 — APPROVED** [review]

**Comment by MeetDOD** [comment]
@VaibhavArora314 @Ultimateutkarsh11 please add level 3 and GGSOC labels please 👍

**Pull request merged**